launch/launch/utilities/signal_management.py

```diff
@@ -12,133 +12,157 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""Module for the signal management functionality."""
+"""Module for signal management functionality."""
 
+import asyncio
+import os
 import platform
 import signal
+import socket
 import threading
 
-import launch.logging
+from typing import Callable
+from typing import Optional
+from typing import Tuple  # noqa: F401
+from typing import Union
 
-__signal_handlers_installed_lock = threading.Lock()
-__signal_handlers_installed = False
-__custom_sigint_handler = None
-__custom_sigquit_handler = None
-__custom_sigterm_handler = None
 
+def is_winsock_handle(fd):
+    """Check if the given file descriptor is WinSock handle."""
+    if platform.system() != 'Windows':
+        return False
+    try:
+        # On Windows, WinSock handles and regular file handles
+        # have disjoint APIs. This test leverages the fact that
+        # attempting to os.dup a WinSock handle will fail.
+        os.close(os.dup(fd))
```

Comment:
is there a better way to test this?

Reply:
I wish there was.

Reply:
maybe https://docs.python.org/3/library/msvcrt.html#msvcrt.get_osfhandle?
otherwise I'm ok with the `os.close(os.dup(fd))` combo

Reply:
That might actually work. In fact, `msvcrt.open_osfhandle` may help to wrap socket handles.

Reply:
> In fact, `msvcrt.open_osfhandle` may help to wrap socket handles.

Scratch that, it doesn't work :) But I did replace `os.dup()` with `msvcrt.get_osfhandle()` in https://github.com/ros2/launch/pull/476/commits/584df2fdcdf84edaedeeb24130c3e086bd3815d0. Thanks !